python/cudf/cudf/core/column/numerical.py

```diff
@@ -597,7 +597,9 @@ def can_cast_safely(self, to_dtype: DtypeObj) -> bool:
                     # TODO: replace np.inf with cudf scalar when
                     # https://github.com/rapidsai/cudf/pull/6297 merges
                     non_infs = s[
-                        ((s == np.inf) | (s == -np.inf)).logical_not()
+                        ((s == np.inf) | (s == -np.inf))
+                        .logical_not()
+                        .fillna(False)
```

Comment:
Why do we need this `fillna`?

Reply:
This is the boolean masking thing as well. If `s` has nulls in it, the resulting mask will also have nulls in it. Then we can't take that object and index into the original `s` with it.

Reply:
I think this goes back up to the boolean masking with `null` values in the mask discussion above.